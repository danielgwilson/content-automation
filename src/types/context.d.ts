export interface IContext {
  outputDir: string;
  resourceDir: string;
  saveOutputToFile: boolean;
  debug: boolean;
}
