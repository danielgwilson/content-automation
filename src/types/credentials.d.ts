export interface ICredentials {
  username: string;
  password: string;
}
