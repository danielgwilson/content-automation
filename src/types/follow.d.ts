export interface IFollowOutput {
  sessionStart: Date;
  sessionEnd: Date;
  context: import("./context").IContext;
  account: string;
  tags: string[];
  actionsTaken: IFollowAction[];
  manager: {
    proxy;
    executablePath;
    timeout;
  };
}

export interface IFollowAction {
  timestamp: Date;
  target: IAccount;
}

export interface IAccount {
  username: string;
  following: number;
  followers: number;
  likes: number;
}
