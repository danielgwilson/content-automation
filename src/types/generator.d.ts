export interface IGeneratorOutput {
  id: string;
  dateGenerated: Date;
  elapsedTime: number;

  media: { metadata: any; thumbnail: any; render: IRenderOutput };
}

export interface IRenderOutput {
  job: any;
  settings: {
    workpath: string;
    maxMemoryPercent: number;
    skipCleanup: boolean;
    debug: boolean;
  };
}
