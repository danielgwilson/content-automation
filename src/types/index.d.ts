export { IContext } from "./context";
export { IPost, IProcessedPost } from "./post";
