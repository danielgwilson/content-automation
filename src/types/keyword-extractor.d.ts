type Language =
  | "danish"
  | "dutch"
  | "english"
  | "french"
  | "galician"
  | "german"
  | "italian"
  | "polish"
  | "portuguese"
  | "romanian"
  | "russian"
  | "spanish"
  | "swedish";

declare module "keyword-extractor" {
  function extract(
    str: string,
    options?: {
      remove_digits: boolean;
      return_changed_case: boolean;
      return_chained_words: boolean;
      language: Language;
      remove_duplicates: boolean;
      return_max_ngrams: boolean;
    }
  ): string[];
  function getStopwords(options: { language?: Language }): string[];
}
