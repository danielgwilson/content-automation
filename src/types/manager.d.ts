interface IManagerOutput {
  proxy?: import("./proxy").IProxy;
  executablePath?: string;
  timeout?: number;
  browserType: string;
}
