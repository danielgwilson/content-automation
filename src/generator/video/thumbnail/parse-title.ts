import fs from "fs";
import path from "path";
import { extract } from "keyword-extractor";

export function parseTitle(title: string, resourceDir: string) {
  const keywords = extract(title);
}

export function getUncommonWords(text: string, resourceDir: string) {
  const words = text.split(" ");
  const commonWords = fs
    .readFileSync(path.join(resourceDir, "/thumbnail/", "common-words.txt"))
    .toString("utf8")
    .split("\n");

  const uncommonWords: string[] = [];
  for (let word of words) {
    if (commonWords.indexOf(word) < 0) uncommonWords.push(word);
  }

  return uncommonWords;
}
