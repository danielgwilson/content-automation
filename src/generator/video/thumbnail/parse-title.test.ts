import { getUncommonWords, parseTitle } from "./parse-title";
import { IContext } from "../../../types";

const context: IContext = {
  outputDir: "./",
  resourceDir: "./resources/",
  saveOutputToFile: true,
  debug: false
};

it("Gets uncommon words in title", () => {
  // const title =
  //   "Would you take a 50/50 chance at $5,000,000 or death? Why or why not?";
  const title =
    "People who haven't pooped in 2019 yet, why are you still holding on to last years shit?";
  const uncommonWords = getUncommonWords(title, context.resourceDir);
  console.log(uncommonWords);
  expect(uncommonWords).toMatchSnapshot();
});

// it("Gets keywords in title", () => {
//   // const title =
//   //   "Would you take a 50/50 chance at $5,000,000 or death? Why or why not?";
//   const title =
//     "People who haven't pooped in 2019 yet, why are you still holding on to last years shit?";
//   const keywords = parseTitle(title, context.resourceDir);
//   console.log(keywords);
//   expect(keywords).toMatchSnapshot();
// });
