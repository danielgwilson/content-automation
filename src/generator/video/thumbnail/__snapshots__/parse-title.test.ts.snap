// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gets uncommon words in title 1`] = `
Array [
  "People",
  "haven't",
  "pooped",
  "2019",
  "yet,",
  "holding",
  "years",
  "shit?",
]
`;
